cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -686,6 +691,8 @@ void deleteApp(Id.Application appId, ApplicationSpecification spec) throws Excep
     }
     deleteProgramLocations(appId);
 
+    deleteAppBusinessMetadata(appId);
```

Comment:
I am not sure what impact that upgrade tool has on this. Is this covered by any test?